stdlib/dynarray.ml

```diff
@@ -131,17 +131,11 @@ let append a b =
 
 let[@inline] get v i =
   if i < 0 || i >= v.size then invalid_arg "Dynarray.get";
-  Array.unsafe_get v.arr i
-
-let[@inline] unsafe_get v i =
-  Array.unsafe_get v.arr i
+  Array.get v.arr i
 
```

Comment:
`Dynarray.unsafe_get/set` are now removed but they make sense in controlled situations (however they are a bit more tricky to use than `Array.unsafe_get/set` due to races). I do not have strong opinions on their inclusion/removal.

Reply:
I think they'd probably be extremely dangerous given the OCaml 5 semantics. However, if the contract is that `unsafe_get` doesn't promise anything wrt memory safety, I suppose the old versions could just work?

Reply:
Could we have a `underlying_array : 'a dynarray -> 'a array` operator instead? This is safe, and it lets people use `unsafe_get` on the resulting array. (For `unsafe_set` it's unclear that users can reason at all on whether the dynarray will update its backing array in the meantime, so probably not a good idea.)

Reply:
Do you mean `underlying_array` would copy? Otherwise, since the array is partially uninitialized/filled with junk, this seems dangerous to me.

Also, if we do that, we should reopen #10982 ? :roll_eyes:

Reply:
Right, `unsafe_to_array` then I guess. (We already have `to_array` that copies in the API, no need for that.)

Re. #10982, I wondered about this as well. I think it's a matter of how confident we are that the implementation will remain a single continuous bytes/array, or may move to another design later. My intuition is that you want to document the fact that dynamic arrays are backed by a single array, and that `Dynarray.get` is morally just as efficient as `Array.get`. Then having a function that breaks this "abstraction" is probably okay. In contract, for buffers it's much less clear that wouldn't want to move to a chunked structure later.

But I'm not insisting on `unsafe_to_array`, I just thought of it as a preferable alternative to `unsafe_get`.

Reply:
A tricky part here is that `unsafe_to_array` might return an array with nullable slots. It's all brand new :). For #10982 it was fine because `bytes` are always valid…